msdsl/generator/verilog.py

```diff
@@ -280,8 +280,13 @@ def make_constant(self, expr: Constant):
         output = Signal(name=next(self.namer), format_=expr.format_)
 
         if isinstance(expr.format_, RealFormat):
-            # compile the range, width, and exponent expressions
-            const = str(expr.value)
+            # compile the range, width, and exponent expressions.
+            if isinstance(expr.value, Integral):
+                # avoid a synthesis corner-case:
+                # https://forums.xilinx.com/t5/Synthesis/Possible-synthesis-bug-casting-integer-to-real-in-a-function/td-p/1140910
+                const = str(float(expr.value))
+            else:
+                const = str(expr.value)
```

Comment:
this solved an annoying synthesis bug in which integer constants were not being represented properly when HARD_FLOAT was defined